utils/generate-commands-json.py

```diff
@@ -0,0 +1,107 @@
+
+    This returns a list with a dict for the command and per each of its
+    subcommands. Each dict contains one key, the command's full name, with a
+    value of a dict that's set with the command's properties and meta
+    information.
+    '''
+    assert len(cmd) >= 8
+    obj = {}
+    rep = [obj]
+    name = cmd[0].upper()
+    meta = cmd[7]
+    key = f'{container} {name}' if container else name
+
+    rep.extend([convert_entry_to_objects_array(name, x)[0] for x in meta.pop('subcommands', [])])
+
+    # The command's value is ordered so the interesting stuff that we care about
+    # is at the start. Optional `None` and empty list values are filtered out.
+    value = OrderedDict()
+    value['summary'] = meta.pop('summary')
+    value['since'] = meta.pop('since')
+    value['group'] = meta.pop('group')
+    set_if_not_none_or_empty(value, 'complexity', meta.pop('complexity', None))
+    set_if_not_none_or_empty(value, 'deprecated_since', meta.pop('deprecated_since', None))
+    set_if_not_none_or_empty(value, 'replaced_by', meta.pop('replaced_by', None))
+    set_if_not_none_or_empty(value, 'history', meta.pop('history', []))
+    set_if_not_none_or_empty(value, 'acl_categories', meta.pop('acl_categories', []))
+    value['arity'] = cmd[1]
+    set_if_not_none_or_empty(value, 'key_specs', [convert_keyspec(x) for x in meta.pop('key_specs',[])])
+    set_if_not_none_or_empty(value, 'arguments', [convert_argument(x) for x in meta.pop('arguments', [])])
+    set_if_not_none_or_empty(value, 'command_flags', [convert_flags_to_boolean_dict(x) for x in meta.pop('command_flags', [])])
+    set_if_not_none_or_empty(value, 'doc_flags', [convert_flags_to_boolean_dict(x) for x in meta.pop('doc_flags', [])])
+    set_if_not_none_or_empty(value, 'hints', meta.pop('hints', []))
+
+    # All remaining meta key-value tuples, if any, are appended to the command
+    # to be future-proof.
+    while len(meta) > 0:
+        (k, v) = meta.popitem()
+        value[k] = v
+
+    obj[key] = value
+    return rep
+
+# MAIN
+if __name__ == '__main__':
+    opts = {
+        'description': 'Transform the output from `redis-cli --json COMMAND` to commands.json format.',
+        'epilog': 'Usage example: src/redis-cli --json COMMAND | utils/generate-commands-json.py'
+    }
+    parser = argparse.ArgumentParser(**opts)
+    parser.add_argument('input', help='JSON-formatted input file (default: stdin)', nargs='?', type=argparse.FileType(), default=stdin)
+    args = parser.parse_args()
+
+    payload = OrderedDict()
+    commands = []
+    data = json.load(args.input)
+
+    for entry in data:
+        cmds = convert_entry_to_objects_array(None, entry)
+        commands.extend(cmds)
+
+    # The final output is a dict of all commands, ordered by name.
+    commands.sort(key=lambda x: list(x.keys())[0])
+    for cmd in commands:
+        name = list(cmd.keys())[0]
+        payload[name] = cmd[name]
+
+    print(json.dumps(payload, indent=4))
```

Comment:
Generally, one more line break will be added here
```suggestion
    print(json.dumps(payload, indent=4))

```